.github/workflows/container-security-scan.yml

```diff
@@ -0,0 +1,117 @@
+# Copyright AGNTCY Contributors (https://github.com/agntcy)
+# SPDX-License-Identifier: Apache-2.0
+name: Container Security Scan
+
+on:
+  push:
+    branches: [ main, "**" ]
+  pull_request:
+  schedule:
+    - cron: '0 3 * * *' # Daily at 03:00 UTC
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  security-events: write # for uploading SARIF
+  actions: read
+
+jobs:
+  trivy-scan:
+    name: Trivy Image Scan (Pull from GHCR)
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - image: dir-apiserver
+            repo: ghcr.io/${{ github.repository_owner }}/dir-apiserver
+            version: latest
+          - image: dir-ctl
+            repo: ghcr.io/${{ github.repository_owner }}/dir-ctl
+            version: latest
```

Comment:
on push and pull request events, these checks will not run on the images produced by that event, but rather whatever latest was pushed. this is because the images are only pushed to `ghcr` after the `CI` pipeline has finished after merging to main (in PRs, we pass it as build artifacts without pushing to ghcr).

If we want to make this work with push and PR events, then we need to wire this into `CI` pipeline properly. For the time being, I would suggest only cron events to trigger this pipeline.

Reply:
No I think we need to have a nightly scan. This is not ready yet. I am triggering the build in PR just to see the output while working on it.